### PR TITLE
Adding a provision test for PRs

### PR DESCRIPTION
**Pull request by fruch**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (`hydra unit-tests`)
- ~~[ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: test-cases/PR-provision-test.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+test_duration: 10
+stress_cmd: ["cassandra-stress write cl=QUORUM duration=1m -schema 'replication(factor=3) compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1000 -pop seq=1..10000000 -log interval=5",
+             "cassandra-stress counter_write cl=QUORUM duration=1m -schema 'replication(factor=3) compaction(strategy=DateTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=5 -pop seq=1..10000000"
+             ]
+
+prepare_write_cmd:  ["scylla-bench -workload=sequential -mode=write -max-rate=300 -replication-factor=3 -partition-count=10 -clustering-row-count=100 -clustering-row-size=5120 -concurrency=7 -rows-per-request=10",
+                     "scylla-bench -workload=uniform -mode=read -replication-factor=3 -partition-count=10 -clustering-row-count=100 -clustering-row-size=5120 -rows-per-request=10 -concurrency=7 -max-rate=32000 -duration=1m"
+                    ]
```
Comment (roydahan) [inline test-cases/PR-provision-test.yaml]: Does it work when you put the read and the write both in the prepare part?

File: test-cases/PR-provision-test.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+n_monitor_nodes: 1
+n_db_nodes: 3
+
+nemesis_class_name: ChaosMonkey
+nemesis_interval: 1
+
+user_prefix: 'PR-provision-test'
+instance_provision: 'spot_low_price'
+
+scylla_version: 3.1.0.rc2
```
Comment (roydahan) [inline test-cases/PR-provision-test.yaml]: Why do you need to provide scylla_version?
Do you want to run it on a fixed version?

Comment (bentsi) [reply test-cases/PR-provision-test.yaml]: we are not checking Scylla but the stability of SCT code. We prefer to keep it constant.

File: test-cases/PR-provision-test.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+prepare_write_cmd:  ["scylla-bench -workload=sequential -mode=write -max-rate=300 -replication-factor=3 -partition-count=10 -clustering-row-count=100 -clustering-row-size=5120 -concurrency=7 -rows-per-request=10",
+                     "scylla-bench -workload=uniform -mode=read -replication-factor=3 -partition-count=10 -clustering-row-count=100 -clustering-row-size=5120 -rows-per-request=10 -concurrency=7 -max-rate=32000 -duration=1m"
+                    ]
+
+n_loaders: 1
+instance_type_db: 'i3.large'
+n_monitor_nodes: 1
+n_db_nodes: 3
+
+nemesis_class_name: ChaosMonkey
```
Comment (bentsi) [inline test-cases/PR-provision-test.yaml]: better to use `NonDisruptiveMonkey`

File: test-cases/PR-provision-test.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+nemesis_interval: 1
+
```
Comment (bentsi) [inline test-cases/PR-provision-test.yaml]: use_mgmt: True

File: test-cases/PR-provision-test.yaml (hunk 1 above)
Comment (fruch) [reply test-cases/PR-provision-test.yaml]: scylla-bench reasults are not checked, so seem like it work, I've copied it from one of the cases using it

File: test-cases/PR-provision-test.yaml (hunk 4 above)
Comment (fruch) [reply test-cases/PR-provision-test.yaml]: done

File: test-cases/PR-provision-test.yaml (hunk 3 above)
Comment (fruch) [reply test-cases/PR-provision-test.yaml]: done

**Comment by bentsi** [comment]
@roydahan can we merge this?

**Pull request merged**